discord_bots/cogs/economy.py

```diff
@@ -0,0 +1,1119 @@
+                return
+
+        # Embed created with player award at index 0
+        embed: Embed = await EconomyCommands.award_currency(interaction, in_progress_game)
+
+        predictions: list[EconomyPrediction] = (
+            session.query(EconomyPrediction)
+            .filter(EconomyPrediction.in_progress_game_id == in_progress_game.id)
+            .all()
+        )
+        # Stop processing if no predictions
+        if len(predictions) == 0:
+            embed.insert_field_at(
+                index=0,
+                name="",
+                value=f"No predictions on game",
+                inline=False
+            )
+        else:
+            winning_team = -1
+            if outcome == "win":
+                winning_team = game_player.team
+            elif outcome == "loss":
+                winning_team = (game_player.team + 1) % 2
+            else:  # tie
+                winning_team = -1
+            # print(f"Wining Team: {winning_team}")
+
+            # Cancel prediction on tie
+            if winning_team == -1:
+                try:
+                    await EconomyCommands.cancel_predictions(interaction, game_id)
+                except ValueError as ve:
+                    # Raised if there are no predictions on this game
+                    embed.insert_field_at(
+                        index=0,
+                        name="Tie game",
+                        value=f"No predictions to be refunded",
+                        inline=False
+                    )
+                    pass
+                except Exception as e:
+                    embed.insert_field_at(
+                        index=0,
+                        name="Tie game",
+                        value=f"Predictions failed to refund: {e}",
+                        inline=False
+                    )
+                    pass
+                else:
+                    embed.insert_field_at(
+                        index=0,
+                        name="Tie game",
+                        value=f"Predictions refunded",
+                        inline=False
+                    )
+                    pass
+                finally:
+                    session.close()
+            else:
+                # Set up winners & losers
+                winning_predictions: list[EconomyPrediction] = []
+                losing_predictions: list[EconomyPrediction] = []
+                for prediction in predictions:
+                    if prediction.team == winning_team:
+                        winning_predictions.append(prediction)
+                    else:
+                        losing_predictions.append(prediction)
+                # print(f"Winning prediction count: {len(winning_predictions)}")
+                # print(f"Losing prediction count: {len(losing_predictions)}")
+
+                # Cancel if either team has no predicitons
+                if len(winning_predictions) == 0 or len(losing_predictions) == 0:               
```

Comment:
```suggestion
                if len(winning_predictions) == 0 or len(losing_predictions) == 0:
```
can you double check if the commit hook is working? theres a script to run the black formatter on the code if you can run that too

Reply:
![image](https://github.com/tribesonecommunity/discord-bots/assets/53620706/c8b2af25-8b40-440f-84a2-51f9252fad9d)
pre-commit should be setup, not sure why it wouldn't be working. I have been formatting as I go, but I think I must have missed the final before PR, have formatted the document to Black again